R/exposure_metrics.R

```diff
@@ -132,3 +132,71 @@ calc_auc_from_regimen <- function(
     target_design$type
   )
 }
+
+#' Calculate time to target attainment
+#'
+#' @param regimen PKPDsim regimen object
+#' @param target_design target design, created using `create_target_design()`
+#' @param auc_comp compartment to look for AUC
+#' @param model PKPDsim model object
+#' @param terminal_interval duration of final dosing interval
+#' @param ... arguments passed on to `PKPDsim::sim`
+#' @return a numeric value indicating the time, in hours, if a dosing interval
+#'   contained the target metric within range. Otherwise, returns NA.
+#' @export
+calc_time_to_target <- function(
+    regimen,
+    target_design,
+    auc_comp,
+    model,
+    terminal_interval = 24,
+    ...
+) {
+  target_type <- match.arg(tolower(target_design$type), mipd_target_types())
```

Comment:
from a user perspective, it might be helpful to check that `target_type` is one supported by this function. At present, we are only returning NA and so it isn't obvious to the user that the reason we are returning NA is due to the target_type simply not being a supported target (versus all their patients failing to attain target)

(the current supported types are any type of AUC, plus trough/cmin)

we currently only use this function for supported targets (except cum_auc i think?, we could write around that or extend it), so I think it would be okay to throw a hard `stop()` for unsupported target types.

if we needed this function to always return a valid number, we could return NA_real_ for unsupported numbers and Inf for never-gets-to-target.